### PR TITLE
PLU-70: [TILES-4] dynamodb mutations

### PR DESCRIPTION
**Pull request by pregnantboy**
3 mutations added here for table rows

- Create table row
- Update table row 
- Delete table row

with tests. also, refactored some test functions into helper functions

Not dealing with graphql schema file yet.

### CODE REVIEW

**Review by liangyuanruo — APPROVED** [review]
LGTM. But I'm familiarising with the codebase at the same time; you may wish for @ogp-weeloong to help take a second pass.

File: packages/backend/src/models/table-metadata.ts [hunk 1]
```diff
@@ -40,6 +40,19 @@ class TableMetadata extends Base {
       },
     },
   })
+
+  async validateRowKeys(data: Record<string, unknown>): Promise<boolean> {
+    const columns = await this.$relatedQuery('columns')
```
Comment (ogp-weeloong) [inline packages/backend/src/models/table-metadata.ts:45]: nit: can limit columns to save some compute. 

```suggestion
    const columns = await this.$relatedQuery('columns').select('id')
```

File: packages/backend/src/graphql/mutations/tiles/create-row.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  params: Params,
+  context: Context,
+) => {
+  const { tableId, data } = params.input
+  const table = await context.currentUser
+    .$relatedQuery('tables')
+    .findById(tableId)
+    .throwIfNotFound()
+
+  if (!(await table.validateRowKeys(data))) {
```
Comment (ogp-weeloong) [inline packages/backend/src/graphql/mutations/tiles/create-row.ts:24]: nit: if you use `withGraphFetched` on the initial query you can potentially save a query here. You can maintain the abstraction by passing `table` into `validateRowKeys`

**Review by ogp-weeloong — APPROVED** [review]
LGTM, tested mutations too

File: packages/backend/src/graphql/mutations/tiles/create-row.ts (hunk 2 above)
Comment (ogp-weeloong) [reply packages/backend/src/graphql/mutations/tiles/create-row.ts:24]: Can do funny thing like this

```
const table = await context.currentUser
    .$relatedQuery('tables')
    .withGraphFetched('columns(onlyColumnId)')
    .modifiers({
      onlyColumnId(builder) {
        builder.select('id')
      },
    })
    .findById(tableId)
    .throwIfNotFound()
```

**Comment by pregnantboy** [comment]
@ogp-weeloong ill leave the microoptimisations for later

**Pull request merged**